swagger.yaml

```diff
@@ -351,30 +347,84 @@ components:
           description: "Network for which to perform statistics request"
       required:
         - network
+    DateParameter:
+      type: string
+      format: date
+      example: "2019-05-17"
+      description: |
+        A starting day for which need to retrieve rewards.
+        This date should be always in UTC time zone.
+    ValidatorIndex:
+      type: integer
+      description: |
+        Validator index on chain
+    EffectivenessRating:
+      type: number
+      description: |
+        Effectiveness rating as reported
+    ValidatorBalance:
+      type: string
+      description: |
+        Balance of validator
+    ValidatorPublicKey:
+      type: string
+      description: |
+        Validator public key, either with or without 0x prefix.
+    ValidatorStatus:
+      type: "string"
+      description: "An optional status filter. Multiple statuses can be requested."
+      enum: ["created", "deposited", "pending", "active", "exited", "withdrawn"]
     ValidatorsRequest:
       type: "object"
       properties:
         from:
-          type: string
-          format: date
-          example: "2019-05-17"
           description: |
```

Comment:
while this is technically correct change, we cannot use this because the code generator doesn't support oneOf/anyOf/allOf

Reply:
the type checking doesn't work properly, and the generator lists these as unsupported https://openapi-generator.tech/docs/generators/rust#schema-support-feature

Reply:
we looks at this with @mksh before and decided to keep the spec this way